### PR TITLE
User source

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

- Track user source

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

**Review by TorstenDittmann — COMMENTED** [review]
Just use something like https://www.npmjs.com/package/js-cookie

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -59,7 +59,32 @@
...
+            const urlParams = $page.url.searchParams;
+            const ref = urlParams.get('ref');
+            const utmSource = urlParams.get('utm_source');
+            const utmMedium = urlParams.get('utm_medium');
+            const utmCampaign = urlParams.get('utm_campaign');
+            let referrer = document.referrer.length ? document.referrer : null;
+            // Skip our own
+            if(referrer?.includes('appwrite.io')) {
+                referrer = null;
+            }
```
Comment (TorstenDittmann) [inline src/routes/+layout.svelte:73]: wouldnt this be useful here?

we have some subdomains leading back to our website

File: src/routes/+layout.svelte [hunk 2]
```diff
@@ -59,7 +59,32 @@
         document.body.classList.add(className);
     }
 
-    onMount(() => {
+    onMount(async () => {
+        if (browser) {
```
Comment (TorstenDittmann) [inline src/routes/+layout.svelte:63]: the browser check is not needed here, rather a check for dev/prod mode so we dont send stuff while developing

File: src/routes/+layout.svelte [hunk 3]
```diff
@@ -59,7 +59,30 @@
...
+        const utmMedium = urlParams.get('utm_medium');
+        const utmCampaign = urlParams.get('utm_campaign');
+        let referrer = document.referrer.length ? document.referrer : null;
+        // Skip our own
+        if(referrer?.includes('//appwrite.io')) {
+            referrer = null;
+        }
+        if (ref || referrer || utmSource || utmCampaign || utmMedium) {
+            try {
+                await createSource(
```
Comment (TorstenDittmann) [inline src/routes/+layout.svelte:75]: shouldn't be awaited 👍🏻

Comment (lohanidamodar) [reply src/routes/+layout.svelte]: fixed

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**